src/utils/GitLabUtils/GitLabUtils.js

```diff
@@ -14,32 +14,47 @@ const getDailyContributions = (gitLabCalendar, date) => {
 
 export const mergeCalendarsContributions = (
   actualCalendar,
-  gitLabUserJsonCalendar
+  gitLabUserJsonCalendar,
+  gitLabUserTimeFrame
 ) => {
   const copiedActualCalendar = javaScriptUtils.deepCopyObject(actualCalendar);
 
   copiedActualCalendar.children[0].children.forEach((weeklyData, weekIndex) => {
     weeklyData.children.forEach((dailyData, dayIndex) => {
-      if (dailyData.attributes["data-count"]) {
-        const actualCalendarDailyData = calendarUtils.getCalendarDataByIndexes(
-          actualCalendar,
-          weekIndex,
-          dayIndex
-        );
-        const totalDailyContributions =
-          Number(actualCalendarDailyData.attributes["data-count"]) +
-          getDailyContributions(
-            gitLabUserJsonCalendar,
-            actualCalendarDailyData.attributes["data-date"]
+      const dayDate = new Date(dailyData.attributes["data-date"]);
+      const timeFrameDateFrom = gitLabUserTimeFrame
+        ? new Date(gitLabUserTimeFrame[0])
+        : dayDate;
+      const timeFrameDateTo = gitLabUserTimeFrame
+        ? new Date(gitLabUserTimeFrame[1])
+        : dayDate;
+
+      if (
+        dailyData.attributes.class === "day" &&
+        dayDate >= timeFrameDateFrom &&
+        dayDate <= timeFrameDateTo
```

Comment:
Let's have a `filter` call based on the config but only if the config is present.

Reply:
Please take a look at the changes. I don't think we need to use `filter()` since the condition is self-explanatory(that is, do *not* skip the day if it's in fact a day and "within" the timeframe if defined).

https://github.com/c-hive/team-contribution-calendar/blob/d711f5012186bc6aee0ab04c1f6e99abb075a5f1/src/utils/GitHubUtils/GitHubUtils.js#L74-L79

Reply:
With a filter, we could avoid one level of nesting. Functional-style code is easier to understand in this case than conditions IMO.

Reply:
Alright. I'll take a look into it. It doesn't seem to be as intuitive as I first thought since the object is quite nested.

Reply:
2d7e08f 👈 This is the "best" I could come up for now. `filterContributionDays()` filters the elements according to certain conditions(e.g. the texts around the calendar are ignored). I wanted to make these nested iterations more "transparent"(e.g. by extracting the weeks from the whole object) but let's address this refactor candidate later.